lib/server-config.js

```diff
@@ -34,28 +37,29 @@ function ensureDirCopyExists (fromDir, toDir) {
  * typically has links to account signup and login, and can be overridden by
  * the server operator.
  *
- * @param argv {Function} Express.js app object
+ * @param argv {Object} Express.js app object
  */
-function ensureWelcomePage (argv) {
-  let { multiuser, templates } = argv
-  let rootDir = path.resolve(argv.root)
-  let serverRootDir
-
-  if (multiuser) {
-    serverRootDir = path.join(rootDir, argv.host.hostname)
-  } else {
-    serverRootDir = rootDir
-  }
+async function ensureWelcomePage (argv) {
+  const { multiuser, templates, server, host, parent } = argv
+  const rootDir = path.resolve(argv.root)
+  const serverRootDir = multiuser ? path.join(rootDir, argv.host.hostname) : rootDir
 
-  let defaultIndexPage = path.join(templates.server, 'index.html')
-  let existingIndexPage = path.join(serverRootDir, 'index.html')
-  let defaultIndexPageAcl = path.join(templates.server, 'index.html.acl')
-  let existingIndexPageAcl = path.join(serverRootDir, 'index.html.acl')
+  // let defaultIndexPage = path.join(templates.server, 'index.html')
+  const existingIndexPage = path.join(serverRootDir, 'index.html')
+  // let defaultIndexPageAcl = path.join(templates.server, 'index.html.acl')
+  // let existingIndexPageAcl = path.join(serverRootDir, 'index.html.acl')
 
   if (!fs.existsSync(existingIndexPage)) {
+    console.log(argv)
     fs.mkdirp(serverRootDir)
-    fs.copySync(defaultIndexPage, existingIndexPage)
-    fs.copySync(defaultIndexPageAcl, existingIndexPageAcl)
+    await fsUtils.copyTemplateDir(templates.server, serverRootDir)
+    await templateUtils.processHandlebarFile(existingIndexPage, {
+      serverName: server ? server.name : host.hostname,
+      serverDescription: server ? server.description : '',
+      serverLogo: server ? server.logo : '',
+      serverVersion: parent._version
```

Comment:
Not sure where `parent` comes from, but just wanted to remark that you can `require('../package.json')` in case you don't.